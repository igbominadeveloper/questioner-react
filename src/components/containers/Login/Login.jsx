import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';

//modules
import { loginUser } from '../../../store/modules/auth';

//components
import Layout from '../../presentationals/Layout';

//stylesheets
import '../../../assets/css/Form.css';

export class Login extends Component {
  state = {
    email: '',
    password: '',
  };

  changeHandler = event => {
    this.setState({
      [event.target.name]: event.target.value,
    });
  };

  submitHandler = event => {
    event.preventDefault();
    const { password, email } = this.state;
    const { from } = this.props.location.state || { from: { pathname: '/' } };
    this.props.loginUser({ email, password }, from);
  };
  render() {
    return (
      <Layout>
        <div className="container">
          <div id="registration-form" className="shadow" data-test="login-form">
            <form onSubmit={this.submitHandler} className="form">
              <p className="form-header">Welcome! Login here</p>
              <div className="form-group">
                <input
                  type="email"
                  className="form-control"
                  name="email"
                  value={this.state.email}
                  onChange={this.changeHandler}
                  id="email"
                  placeholder="Email Address"
                  required
                />
              </div>
              <div className="form-group">
                <input
                  type="password"
                  className="form-control"
                  name="password"
                  value={this.state.password}
                  onChange={this.changeHandler}
                  id="password"
                  placeholder="Password"
                  required
                />
              </div>
              <div className="form-group">
                <button
                  className="form-control btn-primary shadow"
                  data-test="submit-button"
                >
                  Log in
                </button>
              </div>
            </form>
            <p className="form-footer">
              <span>Don't have an account? </span>
              <Link to="/signup" className="form-link text-primary">
                Sign up
              </Link>
            </p>
          </div>
        </div>
      </Layout>
    );
  }
}

const mapStateToProps = state => ({
  isloading: state.auth.isLoading,
  errors: state.auth.errors,
  isAuthenticated: state.auth.isAuthenticated,
});

export default connect(
  mapStateToProps,
  { loginUser },
)(Login);
