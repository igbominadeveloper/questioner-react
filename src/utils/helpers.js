import jwt from 'jsonwebtoken';

export const decodeToken = token => {
  return jwt.decode(token);
};

export const setItem = (name, value) => {
  localStorage.setItem(name, value);
};

export const getItem = name => {
  const item = localStorage.getItem(name);
  return item;
};

export const destroyItem = item => {
  return localStorage.removeItem(item);
};

export const clearLocalStorage = () => {
  return localStorage.clear();
};
