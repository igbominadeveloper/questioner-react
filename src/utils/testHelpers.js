import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

const mockStore = configureStore([thunk]);

export const setupStore = initialState => mockStore(initialState);

export const findByTestAttribute = (component, attribute) => {
  const wrapper = component.find(`[data-test='${attribute}']`);
  return wrapper;
};
