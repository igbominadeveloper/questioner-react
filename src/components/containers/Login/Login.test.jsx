import React from 'react';
import { shallow } from 'enzyme';

import { Login } from './Login';

import { findByTestAttribute } from '../../../utils/testHelpers';

const component = shallow(<Login />);

describe('Custom Search', () => {
  it('renders without errors', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders the form section', () => {
    expect(findByTestAttribute(component, 'login-form').length).toBe(1);
  });

  it('renders a submit button', () => {
    expect(findByTestAttribute(component, 'submit-button').length).toBe(1);
  });
});
