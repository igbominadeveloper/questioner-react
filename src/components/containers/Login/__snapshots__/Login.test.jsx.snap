// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Custom Search renders without errors 1`] = `
<Layout>
  <div
    className="container"
  >
    <div
      className="shadow"
      data-test="login-form"
      id="registration-form"
    >
      <form
        className="form"
        onSubmit={[Function]}
      >
        <p
          className="form-header"
        >
          Welcome! Login here
        </p>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            required={true}
            type="email"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            required={true}
            type="password"
            value=""
          />
        </div>
        <div
          className="form-group"
        >
          <button
            className="form-control btn-primary shadow"
            data-test="submit-button"
          >
            Sign up
          </button>
        </div>
      </form>
      <p
        className="form-footer"
      >
        <span>
          Don't have an account? 
        </span>
        <Link
          className="form-link text-primary"
          to="/signup"
        >
          Sign up
        </Link>
      </p>
    </div>
  </div>
</Layout>
`;
